assertsprocessor/processor.go

```diff
@@ -0,0 +1,70 @@
+package assertsprocessor
+
+import (
+	"context"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+// The methods from a Span that we care for to enable easy mocking
+
+type assertsProcessorImpl struct {
+	logger           *zap.Logger
+	config           *Config
+	nextConsumer     consumer.Traces
+	thresholdsHelper *thresholdHelper
+	metricBuilder    *metricHelper
+	sampler          *sampler
+}
+
+// Capabilities implements the consumer.Traces interface.
+func (p *assertsProcessorImpl) Capabilities() consumer.Capabilities {
+	p.logger.Info("consumer.Capabilities callback")
+	return consumer.Capabilities{MutatesData: false}
+}
+
+// Start implements the component.Component interface.
+func (p *assertsProcessorImpl) Start(ctx context.Context, host component.Host) error {
+	p.logger.Info("consumer.Start callback")
+	err := p.metricBuilder.compileSpanFilterRegexps()
```

Comment:
> nit: Go's style is put `(err != nil)` in the `if` statement, so this would be
> 
> ```
> 	if err := p.metricBuilder.compileSpanFilterRegexps(); err != nil {
> 		return err
> 	}
> ```
> 
> IDE will collapse this, and then the following code does not need to be indented inside another {}.

Done